### PR TITLE
Update name

### LINKED ISSUE #96: Auto-format names
Changes to be made:

Should check be case INSENSITIVE, only alphabet and space and auto format to first letter is caps (stored in value)

### PR DESCRIPTION
**Pull request by TY1Fan**
Closes #96 

Changes:
Added a formatter to name class.
Will format names with different capitalization.
E.g. BeTSY CroWES will be converted and stored as Betsy Crowes

@DennieDan I have changed the test cases as stated in your previous pull request

### CODE REVIEW

File: src/main/java/seedu/address/model/person/Name.java [hunk 1]
```diff
@@ -38,6 +38,21 @@ public static boolean isValidName(String test) {
         return test.matches(VALIDATION_REGEX);
     }
 
+    private static String formatName(String name) {
+        String[] words = name.toLowerCase().split("\\s+");
+        StringBuilder formattedWords = new StringBuilder();
+
+        for (String word : words) {
+            if (!word.isEmpty()) {
+                formattedWords.append(Character.toUpperCase(word.charAt(0)))
```
Comment (Feng1231) [inline src/main/java/seedu/address/model/person/Name.java:47]: I think this block of code need to be abstracted out, it took me quite long to understand what this piece of code is doing

**Review by Feng1231 — CHANGES_REQUESTED** [review]
Nice, testing done too! But a piece of code needs to be abstracted out

File: src/main/java/seedu/address/model/person/Name.java (hunk 1 above)
Comment (TY1Fan) [reply src/main/java/seedu/address/model/person/Name.java]: Thanks for the suggestion! Ive refactored the code

File: src/main/java/seedu/address/model/person/Name.java [hunk 2]
```diff
@@ -38,6 +38,30 @@ public static boolean isValidName(String test) {
...
 
+    private static String formatName(String name) {
+        String[] words = name.toLowerCase().split("\\s+");
+        StringBuilder formattedWords = new StringBuilder();
+
+        for (String word : words) {
+            if (!word.isEmpty()) {
+                formattedWords.append(capitalizeWord(word)).append(" ");
+            }
+        }
```
Comment (DennieDan) [inline src/main/java/seedu/address/model/person/Name.java:49]: Looks better and easier to understand

**Review by DennieDan — APPROVED** [review]
LGTM

File: src/main/java/seedu/address/model/person/Name.java (hunk 2 above)
Comment (DennieDan) [reply src/main/java/seedu/address/model/person/Name.java:49]: You wna abstract the loop also? To look neater

Comment (TY1Fan) [reply src/main/java/seedu/address/model/person/Name.java:60]: Thank you for the suggestion! Ive abstracted the initial format name into 3 different methods that better reflects the process of formatting the name.

**Review by Feng1231 — APPROVED** [review]
LGTM

**Review by teamotitan — APPROVED** [review]
LGTM

**Pull request merged**